tasks/src/feature-extraction/about.md

```diff
@@ -0,0 +1,126 @@
+
+Taking top 100 rows from Newsgroup classification data from Sklearn
+
+```python
+dataset =  datasets.fetch_20newsgroups()
+X = dataset.data[:100]
+y = dataset.target[:100]
+```
+This function creates a sparse embedding for each token present in the sentence. Please remember to use the same tokenizer that you would be using while calling the `feature-extraction` pipeline. As the default, BERT tokenizer used in the pipeline is bert-base-cased, so we use the same for initializing the dataframe with number of columns equal to vocabulary size of the tokenizer. We are using a dataframe within the function for ease of understanding, and it returns an `np.array` of size `(n_samples,vocab_size)`.
+
+```python
+def get_features(X,feature_extraction):
+
+    # Initializing the input dataframe
+    tokenizer = AutoTokenizer.from_pretrained('bert-base-cased')
+    df = pd.DataFrame(columns = sorted(tokenizer.vocab.values()))
+
+    # Looping over the input data
+    for i in tqdm(X):
+        tokenized_input = tokenizer(i)
+
+        # Getting embeddings for each token
+        features = feature_extraction(input_)
+        features = np.squeeze(features,axis = 0)
+
+        # Calculating mean of embeddings for each token over 768 embedding values
+        features = np.mean(features,axis = 1)
+
+        # Mapping token embeddings to their specific column in the data-frame
+        features_dict = dict(zip(tokenized_input['input_ids'],features))
+        df = df.append(features_dict, ignore_index=True)
+
+    # Filling NaN values with 0
+    df = df.fillna(0)
+    return df.values
+```
+Below we perform a 80:20 train-test split for modeling
+
+```python
+X_features = get_features(X,feature_extraction)
+X_train, X_test, y_train, y_test = train_test_split(X_features, y,stratify = y, test_size=0.2, random_state=42)
+```
+
+Performing Multi-class classification using sklearn Random Forest
+
+```python
+model = RandomForestClassifier(random_state=42)
+model.fit(X_train,y_train)
+y_pred = model.predict(X_test)
+print(f'Test Accuracy is {accuracy_score(y_test,y_pred)*100} %')
+
+# Output:
+# Test Accuracy is 95.0 %
+```
+This exercise shows the potential of using embeddings as input to a classifier. For an imbalanced dataset, we can use performance metrics like F1-score and ROC AUC.
+
+
+# Task Variants
+
+## Inference
+
+This pipeline extracts the hidden states from the base transformer, which can be used as features in downstream tasks. You can infer the text data with the 🤗 Transformers library using the `feature-extraction` pipeline. This pipeline takes a string or a list of string input and generates features for each input. All models may be used for this pipeline (see a list of all models on https://huggingface.co/models).
+
+The output generated for a single input is a shape of a dataframe, which contains 768 embeddings for each token. See the use-case above, where we calculate mean of these embeddings and use them as input to a classifier.
+
+```python
```

Comment:
In this part it's best if we only show how to infer and use this pipeline for semantic similarity, does it sound good? Given you've put a lot of effort into this code, maybe we could simply mention this as another way to use this pipeline. We want to enable readability for beginner audience. Thanks a lot for your patience!